src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/builders/AbstractTemplateBuilder.java

```diff
@@ -94,7 +94,6 @@ public int getFilterContextPos() {
     protected void writeKey(TemplateOutputWriter writer) throws IOException {
         if (key != null && !key.evaluate(null).equals(""))
             writer.writeElementName(key.evaluate(null));
-        else throw new RuntimeException("Cannot write null key value");
```

Comment:
So we can write a ``NULL`` key?

Reply:
evaluating on null since key might be an env function or a literal. In both cases the evaluate(Object o) method doesn't need any argument